future-apps/payroll/contracts/Payroll.sol

```diff
@@ -634,4 +668,48 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
 
         return xrt;
     }
+
+    /**
+     * @dev Loop over tokens to send requested amount to the employee
+     * @param _employeeId Employee's identifier
+     * @param _totalAmount Total amount to be transferred to the employee distributed through the setup tokens allocation.
+     * @param _reference String detailing payment reason.
+     * @return True if there was at least one token transfer.
+     */
+    function _transferTokensAmount(uint256 _employeeId, uint256 _totalAmount, string _reference) private returns (bool somethingPaid) {
+        Employee storage employee = employees[_employeeId];
+        for (uint256 i = 0; i < allowedTokensArray.length; i++) {
+            address token = allowedTokensArray[i];
+            if (employee.allocation[token] != uint256(0)) {
+                uint128 exchangeRate = _getExchangeRate(token);
+                require(exchangeRate > 0, ERROR_EXCHANGE_RATE_ZERO);
+                // Salary converted to token and applied allocation percentage
+                uint256 tokenAmount = _totalAmount.mul(exchangeRate).mul(employee.allocation[token]);
+                // Divide by 100 for the allocation and by ONE for the exchange rate
+                tokenAmount = tokenAmount / (100 * ONE);
+                finance.newPayment(token, employee.accountAddress, tokenAmount, 0, 0, 1, _reference);
+                emit SendPayment(employee.accountAddress, token, tokenAmount, _reference);
+                somethingPaid = true;
+            }
+        }
+    }
+
+    /**
+     * @dev Try removing employee if there are no pending payments and has reached employee's end date.
+     * @param _employeeId Employee's identifier
+     * @param _date Date timestamp used to evaluate if the employee can be removed from the payroll.
```

Comment:
I don't think we should pass this in as a parameter, but always use `getTimestamp64()`.

Reply:
I see your point but I think it makes sense to use the min date between current timestamp and employee end-date, because we would get an invalid owed salary amount otherwise. That's why I parameterized it and make it `private`. 

I was also thinking that maybe we could move the `min(timestamp, enddate)` logic within the owed salary amount calc since we will need to use it always as is I think.

Reply:
Good point, but then I would do something like:
```
if (employee.endDate > getTimestamp64()) {
    return;
}
if (_getOwedSalary(_employeeId, employee.endDate) > 0) {
   return;
}
if (employee.accruedValue > 0) {
   return;
}

delete employeeIds[employee.accountAddress];
delete employees[_employeeId];
```
This way we are also saving some gas in most cases (where employee didn't reach the end date, which should be the regular case)

Reply:
I finally solve it like [this](https://github.com/aragon/aragon-apps/pull/744/commits/8a86879ea0f9bccea8a3f682c41b9d09ba769726). As you can see I decided to move the `min(current, end date)` thing within `_getOwedSalary` since it was actually misleading other calculations for partial payrolls.

Reply:
Sounds good!